src/multivariate/jointorderstatistics.jl

```diff
@@ -0,0 +1,132 @@
+# Implementation based on chapters 2-4 of
+# Arnold, Barry C., Narayanaswamy Balakrishnan, and Haikady Navada Nagaraja.
+# A first course in order statistics. Society for Industrial and Applied Mathematics, 2008.
+
+"""
+    JointOrderStatistics <: ContinuousMultivariateDistribution
+
+The joint distribution of a subset of order statistics from a sample from a continuous univariate distribution.
+
+    JointOrderStatistics(
+        dist::ContinuousUnivariateDistribution,
+        n::Int,
+        r::AbstractVector{Int}=1:n;
+        check_args::Bool=true,
+    )
+
+Construct the joint distribution of order statistics `r` from an IID sample of size `n` from `dist`.
+
+The ``i``th order statistic of a sample is the ``i``th element of the sorted sample.
+For example, the 1st order statistic is the sample minimum, while the ``n``th order
+statistic is the sample maximum.
+
+`r` must be a sorted vector of unique integers between 1 and `n`.
+
+For a single order statistic, use [`OrderStatistic`](@ref) instead.
+
+## Examples
+
+```julia
+JointOrderStatistics(Normal(), 10)           # Product(fill(Normal(), 10)) restricted to ordered vectors
+JointOrderStatistics(Cauchy(), 10, [1, 10])  # joint distribution of the extrema
+```
+"""
+struct JointOrderStatistics{D<:ContinuousUnivariateDistribution,R<:AbstractVector{Int}} <:
```

Comment:
For the continuous case, we can write down the PDF and not much else (except the mean and covariance for a handful of `dist`s, for a future PR).

For the discrete case, the PDF is much more complicated and probably only tractable when `length(r) <= 2`.  For a discrete `dist` whose support is a finite set, we can define a distribution of the order statistics for a sample _without_ replacement, at least in the bivariate case. The PMF, mean, and covariance are known. I wonder if that should be a special case of this distribution.

Reply:
> I wonder if that should be a special case of this distribution.

Nah, seems complicated and not all that useful.